x-pack/plugin/otel-data/build.gradle

```diff
@@ -34,4 +42,100 @@ dependencies {
   clusterModules project(xpackModule('stack'))
   clusterModules project(xpackModule('wildcard'))
   clusterModules project(xpackModule('mapper-version'))
+
+  implementation "com.google.protobuf:protobuf-java:${protobufVersion}"
+}
+
+protobuf {
+  protoc {
+    // The artifact spec for the Protobuf Compiler
+    artifact = "com.google.protobuf:protoc:${protobufVersion}"
+  }
+}
+
+def otlpProtoVersion = "1.7.0"
+
+// To generate the checksum, download the file and run "shasum -a 256 ~/path/to/opentelemetry-proto-<version>.zip"
+def protoChecksum = "ddb80357ff146f5e3bda584907185b1f635412a4b31edf6f96b102a18b8e05dc"
+def protoArchivePath = layout.buildDirectory.file("archives/opentelemetry-proto-${otlpProtoVersion}.zip")
+
+def downloadProtoArchive = tasks.register("downloadProtoArchive") {
+  outputs.file(protoArchivePath)
+  onlyIf { protoArchivePath.get().asFile.exists() == false }
+  doLast {
+    protoArchivePath.get().asFile.parentFile.mkdirs()
+    "https://github.com/open-telemetry/opentelemetry-proto/archive/v${otlpProtoVersion}.zip".toURL()
+      .withInputStream { is ->
+        protoArchivePath.get().asFile.withOutputStream {os -> os << is }
+      }
+  }
+}
+
+def verifyProtoArchive = tasks.register("verifyProtoArchive") {
+  dependsOn(downloadProtoArchive)
+  doLast {
+    protoArchivePath.get().asFile.withInputStream { inputStream ->
+      def sha256 = inputStream.readAllBytes().digest("SHA-256")
+      if (sha256 != protoChecksum) {
+        throw new GradleException("Checksum verification failed for $protoArchivePath: expected $protoChecksum, got $sha256")
+      }
+    }
+  }
+}
+
+def unzipProtoArchive = tasks.register("unzipProtoArchive", Copy) {
+  dependsOn(verifyProtoArchive)
+  from zipTree(protoArchivePath)
+  into layout.buildDirectory.dir("protos")
+}
+
+// Avoid unnecessary String allocations in the generated AnyValue class
+// We always need the ByteString (UTF-8) representation of string attributes
+def adjustAnyValueBuilder = tasks.register("adjustAnyValueBuilder") {
+  doLast {
+    ant.replace(
```

Comment:
We are in the process of removing all usages of antBuilder in our build scripts as it has a couple of flaws and breaks new features in gradle.